### PR TITLE
feat(semantic-release): add semantic release to the project

### PR DESCRIPTION
**Pull request by luabida**
I've also noticed `requirements.txt` and a `setup.py` file, I chose to use the poetry to do the build, since setuptools is not as dynamic

### CODE REVIEW

**Comment by luabida** [comment]
I'm not sure if `PYPI_TOKEN` is present in the secrets, I don't have access to check it out

**Comment by xmnlab** [comment]
@fccoelho is the only maintainer there: https://pypi.org/project/pysus/
@fccoelho could you generate a token there and create a secret here called PYPI_TOKEN with that token pls?

**Comment by xmnlab** [comment]
in general looks good to me.
just one thing, could you add to the release workflow the options for PR and push?

you can take a look at here https://github.com/arxlang/arx/blob/main/.github/workflows/release.yaml#L6

check that some steps has `if` statements specific for workflow or for "not" workflow

File: pysus/tests/test_data/test_sinan.py [hunk 1]
```diff
@@ -99,10 +98,6 @@ def test_type_convertion(self):
         self.assertTrue(self.dbf_name.exists())
         assert not all(df.dtypes == "object")
 
-    def test_geocode(self):
-        self.assertTrue(self.data_pickle.exists())
```
Comment (luabida) [inline pysus/tests/test_data/test_sinan.py:103]: I didn't see much use for this test..

Comment (esloch) [reply pysus/tests/test_data/test_sinan.py:103]: I don't remember if at some point sinan generates this pickle file, but from the name of the test it seems that this test is incomplete.

File: pysus/tests/test_utilities.py [hunk 2]
```diff
@@ -18,15 +18,13 @@ class TestReadDbc(unittest.TestCase):
     dbf_fname = TEST_DATA / "EPR-2016-06-01-2016.dbf"
     dbc_fname = TEST_DATA / "sids.dbc"
 
+    @unittest.skip('Issue #111')
     def test_read_dbc(self):
         df = read_dbc(str(self.dbc_fname))
         self.assertIsInstance(df, pd.DataFrame)
         self.assertGreater(df.size, 0)
 
-    def test_read_dbc_dbf(self):
```
Comment (luabida) [inline pysus/tests/test_utilities.py:26]: I've split this test and remove the duplicated part, that is now skipped

**Review by esloch — APPROVED** [review]
LGTM!

**Pull request merged**